internal/datastore/crdb/watch.go

```diff
@@ -119,6 +120,17 @@ func (cds *crdbDatastore) Watch(ctx context.Context, afterRevision datastore.Rev
 				})
 
 				for _, change := range toEmit {
+					for int(cds.watchBufferLength) <= len(updates) {
```

Comment:
So I'm not convinced this is necessary; if we remove the `default` clause from the select below, then the `updates <- change` should hang if the update channel is filled.

Can you try that approach?

Reply:
@staskosatkin

Reply:
Correct; the channel insert will hang, which is what we expect, correct?

Reply:
I will check this approach and update this PR after that. Thank you.

Reply:
@staskosatkin Checking in on this

Reply:
@staskosatkin Checking in again